systemvm/debian/opt/cloud/bin/cs/CsAddress.py

```diff
@@ -554,7 +554,7 @@ def fw_vpcrouter(self):
             if self.address["source_nat"]:
                 self.fw.append(["nat", "front",
                                 "-A POSTROUTING -o %s -j SNAT --to-source %s" %
-                                (self.dev, self.address['public_ip'])])
+                                (self.address['device'], self.address['public_ip'])])
```

Comment:
this line applies only when private gateway is source nat.

it seems we need to change line 698-700

https://github.com/apache/cloudstack/pull/9552/files#diff-3c470eee70094a82ad3ed790deed16a991e75ed18901cfb82d82a80cd71228a7L698-R700

Reply:
ok, I'll try and find if we have data on the second IF at that point.

Reply:
@DaanHoogland 
have you checked the new iptables rules ? do they look good ?

Reply:
no they don't. I'll need to find the origin of the data involved and analyse the logs and processed json files in the VR to see what happens.

Reply:
try changes on line 698-700 
@DaanHoogland

Reply:
have a look at those @weizhouapache , they work in my test.

Reply:
it looks like this code snippet (line 554 to 557) can be removed.
It has been covered by line 696-697 (new code)

Reply:
please ignore my previous comment

Reply:
need to check, for private gateway , is `self.dev` same as `self.address['device']` ?